warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -463,6 +465,15 @@ public void seek(Range r, Collection<ByteSequence> columnFamilies, boolean inclu
             // the start of this range is beyond the end of the last range seeked
             // we must reset keyValues to null and empty the underlying collection
             clearRowBasedHdfsBackedSet();
+        } else {
+            // inside the original range, so potentially need to reposition keyValues
+            if (keyValues != null) {
+                Key startKey = r.getStartKey();
+                // decide if keyValues needs to be rebuilt or can be reused
+                if (!keyValues.hasNext() || (keyValues.peek().getKey().compareTo(startKey) > 0)) {
```

Comment:
Do we need to peek here or can we just use topKey?

Reply:
down below in findTop() next() is called at least once so we have to check the peek value instead of the curren tvalue